tests/pytests/common.py

```diff
@@ -320,14 +320,17 @@ def get_vecsim_debug_dict(env, index_name, vector_field):
     return to_dict(env.cmd(debug_cmd(), "VECSIM_INFO", index_name, vector_field))
 
 
-def forceInvokeGC(env, idx = 'idx', timeout = None):
+def forceInvokeGC(env, idx='idx', timeout=None, run_in_background=False):
```

Comment:
Does the scenario you describe, where the child process is starved, might occur in all the tests, particularly when we run in sanitizer mode?
If so consider setting `run_in_background`default value to `True`

Reply:
But other tests do rely on running the GC *not* in the background